### PR TITLE
KAZOO-3497: When presence_id is updated it also update it in the provisi...

### PR DESCRIPTION
**Pull request by macpie**
...oner

### CODE REVIEW

**Comment by macpie** [comment]
Tested with provisioner, device presence id was successfully updated

File: applications/crossbar/priv/couchdb/account/devices.json [hunk 1]
```diff
@@ -13,6 +13,9 @@
         },
         "listing_by_macaddress": {
             "map": "function (doc) {if (doc.pvt_type != 'device' || doc.pvt_deleted) return; if(doc.mac_address) {emit(doc.mac_address, null)}}"
+        },
+        "listing_by_presence": {
```
Comment (jamesaimonetti) [inline applications/crossbar/priv/couchdb/account/devices.json]: This name suggests, to me, the presence status of the device (ringing, answered, hungup). What other names would be more descriptive of what this view is showing?

File: applications/crossbar/src/modules_v2/cb_users_v2.erl [hunk 2]
```diff
@@ -262,6 +265,47 @@ patch(Context, _Id) ->
...
+    Doc = cb_context:doc(Context),
+    case wh_json:get_value(<<"presence_id">>, DbDoc) =:= wh_json:get_value(<<"presence_id">>, Doc) of
+        'true' ->
+            lager:debug("presence_id did not change, ignoring");
+        'false' ->
+            update_devices_presence(Context)
+    end.
+
+
+update_devices_presence(Context) ->
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules_v2/cb_users_v2.erl]: Spec

File: applications/crossbar/priv/couchdb/account/devices.json [hunk 3]
```diff
@@ -13,6 +13,9 @@
...
+            "map": "function(doc) {if (doc.pvt_type != 'device' || doc.pvt_deleted) return;if (doc.provision && doc.provision.feature_keys) {for(var i in doc.provision.feature_keys) {var key = doc.provision.feature_keys[i];if(key.type == 'presence' || key.type == 'personal_parking') {emit(key.value, key.type);}}}}"
```
Comment (jamesaimonetti) [inline applications/crossbar/priv/couchdb/account/devices.json]: You don't use the "value" (`key.type`) below. Any reason not to just do `emit(key.value, null)` ? Would save space on the DB servers.

File: applications/crossbar/src/modules_v2/cb_users_v2.erl [hunk 4]
```diff
@@ -233,7 +234,9 @@ validate(Context, UserId, ?QUICKCALL, _) ->
 -spec post(cb_context:context(), path_token()) -> cb_context:context().
 post(Context, _) ->
     _ = crossbar_util:maybe_refresh_fs_xml('user', Context),
-    crossbar_doc:save(Context).
+    Context1 = crossbar_doc:save(Context),
+    _  = maybe_update_devices_presence(Context),
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules_v2/cb_users_v2.erl]: Shouldn't you test `cb_context:resp_status(Context1)` here for success. No need to update if the save failed!

**Comment by jamesaimonetti** [comment]
Merged manually, due to needing rebase.

**Pull request closed without merging**